### PR TITLE
[Nano] add pytorch inference doc

### PR DESCRIPTION
**Pull request by zhentaocc**

### CODE REVIEW

**Comment by zhentaocc** [comment]
@yangw1234 pls take a look. Maybe anyone who is working on the notebook and examples can help and see if anything improper.

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_inference.md [hunk 1]
```diff
@@ -1,8 +1,215 @@
...
-add a link for examples here.
+BigDL-Nano provides serveral APIs which can easily apply optimizations on inference pipeline to improve latency and througput. Currently, performance accelerations are achieved by integrating extra runtime as inference backend engine or using quantization methods on full-precision trained models to reduce computation during inference. Trainer (`bigdl.nano.pytorch.Trainer`) provides the APIs for all optimizations you need for inference.
 
-###  Runtime Acceleration
-onnx runtime, openvino
+For runtime acceleration, BigDL-Nano has enabled two kinds of runtime for users in `Trainer.trace()`, ONNXRuntime and OpenVINO.
 
-### Quantization
\ No newline at end of file
+For quantization, BigDL-Nano provides only post-training quantization `trainer.quantize()` for users to inference with models of 8-bit precision. Quantization-Aware Training is not available for now. Model conversion to 16-bit like BF16, FP16 will be coming soon.
```
Comment (yangw1234) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_inference.md:7]: "provides only post-training quantization `trainer.quantize()`" -> "provides only post-training quantization in `trainer.quantize()`"

Are we going to support FP16?

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_inference.md [hunk 2]
```diff
@@ -1,8 +1,215 @@
...
+...
+...
+
+# Inference/Prediction
+trainer.validate(ort_model, dataloader)
+trainer.test(ort_model, dataloader)
+trainer.predict(ort_model, dataloader)
+```
+### ONNXRuntime Acceleration
+Before you start with any BigDL-Nano usage, you are required to install some onnx packages as follow to setup your environment with ONNXRuntime acceleration.
```
Comment (yangw1234) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_inference.md:40]: "Before you start with any BigDL-Nano usage" might be confusing. How about "Before you start with onnxruntime accelerator"?

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_inference.md [hunk 3]
```diff
@@ -1,8 +1,215 @@
...
+trainer.predict(ort_model, dataloader)
+# note that `ort_model` is not trainable any more, so you can't use like 
+# trainer.fit(ort_model, dataloader) # this is illegal
+```
+### OpenVINO Acceleration
+To use OpenVINO acceleration, you have to install openvino toolkit:
+```shell
+pip install openvino-dev
+```
+The OpenVINO usage is quite similar to ONNXRuntime, following usage is for openvino:
```
Comment (yangw1234) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_inference.md:66]: "following usage" -> "the following usage"?

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_inference.md [hunk 4]
```diff
@@ -1,8 +1,215 @@
...
+trainer = Trainer()
+trainer.validate(ort_model, dataloader)
+trainer.test(ort_model, dataloader)
+trainer.predict(ort_model, dataloader)
+# note that `ort_model` is not trainable any more, so you can't use like 
+# trainer.fit(ort_model, dataloader) # this is illegal
+```
+
+## Quantization
+Quantization is widely used to compress models to lower precision which not only reduces the model size but also accelerates inference speed. BigDL-Nano provides `Trainer.quantize()` API for users to quickly obtain a quantized model with accuracy control by specifying a few arguments. Intel Neural Compressor (INC) and Post-training Optimization Tools (POT) from OpenVINO toolkit are enabled as options. At the meantime, runtime acceleration is also included directly in quantization pipeline when using accelerator='onnxruntime'/'openvino' so you don't have to run `Trainer.trace` before quantization.
```
Comment (yangw1234) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_inference.md:86]: "to compress models to lower precision" -> "to compress models to a lower precision"?
"to lower precision which not only" -> "to lower precision, which not only"?
"accelerates inference speed" -> "accelerates inference" or "improve inference speed"?

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_inference.md [hunk 5]
```diff
@@ -1,8 +1,215 @@
 # BigDL-Nano PyTorch Inference Overview
 
-add a link for examples here.
+BigDL-Nano provides serveral APIs which can easily apply optimizations on inference pipeline to improve latency and througput. Currently, performance accelerations are achieved by integrating extra runtime as inference backend engine or using quantization methods on full-precision trained models to reduce computation during inference. Trainer (`bigdl.nano.pytorch.Trainer`) provides the APIs for all optimizations you need for inference.
```
Comment (yangw1234) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_inference.md:3]: Maybe "which can be easily used to apply optimizations"? Feels weird to say "APIs apply optimizations".

"on inference pipeline" -> "on inference pipelines" ?

"extra runtime as inference backend engine" -> "extra runtimes as inference backend engines" ?

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_inference.md [hunk 6]
```diff
@@ -1,8 +1,215 @@
...
+
+To use INC as your quantization engine, you can choose accelerator as None or 'onnxruntime'. Otherwise, accelerator='openvino' means using OpenVINO POT to do quantization.
+
+By default, `Trainer.quantize()` doesn't search the tuning space and returns the fully-quantized model without considering the accuracy drop. If you need to search quantization tuning space for a model with accuracy control, you'll have to specify a few arguments to define the tuning space. More instructions in [Quantization with Accuracy control](#quantization-with-accuracy-control)
+
+### Quantization using Intel Neural Compressor
+By default, Intel Neural Compressor is not installed with BigDL-Nano. So if you determine to use it as your quantization backend, you'll need to install it first:
+```shell
+pip install neural-compressor
```
Comment (yangw1234) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_inference.md:95]: Should we specify version?

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_inference.md [hunk 7]
```diff
@@ -1,8 +1,215 @@
...
+
+# validate, predict, test in Trainer also support acceleration
+trainer.validate(q_model, dataloader)
+trainer.test(q_model, dataloader)
+trainer.predict(q_model, dataloader)
+```
+This is a most basic usage to quantize a model with defaults, INT8 precision and without search tuning space to control accuracy drop.  
+
+**Quantization with ONNXRuntime accelerator**  
+Without ONNXRuntime accelerator, `Trainer.quantize()` will return a model with compressed precision but runing inference in ONNXRuntime engine. If your INC version >= 1.11, it's also required to install onnxruntime-extensions as a dependency of INC when using ONNXRuntime as backend as well as the dependencies required in [ONNXRuntime Acceleration](#onnxruntime-acceleration):
```
Comment (yangw1234) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_inference.md:112]: "but runing inference" -> "but running inference"

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_inference.md [hunk 8]
```diff
@@ -1,8 +1,215 @@
...
+
+# run inference with transparent acceleration 
+y_hat = ort_q_model(x)
+trainer.validate(ort_q_model, dataloader)
+trainer.test(ort_q_model, dataloader)
+trainer.predict(ort_q_model, dataloader)
+```
+
+### Quantization using Post-training Optimization Tools
+Post-training Optimization Tools is provided by OpenVINO toolkit. To use POT, you need to install OpenVINO same in [OpenVINO acceleration](#openvino-acceleration):
```
Comment (yangw1234) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_inference.md:140]: "install OpenVINO same in " -> "install OpenVINO as the same in "?

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_inference.md [hunk 9]
```diff
@@ -1,8 +1,215 @@
...
+```shell
+pip install openvino-dev
+```
+Take the example in [Runtime Acceleration](#runtime-acceleration), add quantization:
```
Comment (yangw1234) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_inference.md:144]: "add quantization" -> "and add quantization"

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_inference.md [hunk 10]
```diff
@@ -1,8 +1,215 @@
...
+ov_q_model = trainer.quanize(model, accelerator='openvino', calib_dataloader=dataloader)
+# run simple prediction with transparent acceleration
+y_hat = ov_q_model(x)
+
+# validate, predict, test in Trainer also support acceleration
+trainer.validate(ov_q_model, dataloader)
+trainer.test(ov_q_model, dataloader)
+trainer.predict(ov_q_model, dataloader)
+```
+Same as ONNXRuntime, it equals to converting the model from Pytorch to OpenVINO firstly and then do quantization on the converted OpenVINO model:
```
Comment (yangw1234) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_inference.md:155]: "then do quantization on" -> "then doing quantization on"

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_inference.md [hunk 11]
```diff
@@ -1,8 +1,215 @@
...
+- `metric`: A metric of `torchmetric` to run evaluation and compare with baseline
+
+- `accuracy_criterion`: A dictionary to specify the acceptable accuracy drop, e.g. `{'relative': 0.01, 'higher_is_better': True}`
+
+    - `relative` / `absolute`: Drop type, the accuracy drop should be relative or absolute to baseline
+    - `higher_is_better`: Indicate if larger value of metric means better accuracy
+- `max_trials`: Maximum trails on the search, if the algorithm can't find a satisfying model, it will exit and raise error.
+
+**Accuracy Control with INC**
+There are a few arguments that requires only by INC, don't specify or modify any of them if you use `accelerator='openvino'`.
```
Comment (yangw1234) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_inference.md:180]: "don't specify or modify" -> "and you should not"?

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_inference.md (hunk 1 above)
Comment (zhentaocc) [reply docs/readthedocs/source/doc/Nano/QuickStart/pytorch_inference.md]: yes. we will support FP16.
